### PR TITLE
Issue/1888 remove depracated option variable iso5

### LINKED ISSUE #1888: remove support for leaving nullable attribute unassigned
Currently nullable attributes are represented by `OptionVariable`. They should instead be represented by a plain `ResultVariable` with a nullable type. See #1775

- [x] Fix std
- [ ] Fix vyos
- [x] Ask solutions team to look into vcenter and yang (tickets created by solutions team)
- [ ] Implement the change

this is only for iso6

This a grep report from the jenkins jobs:
[https://app.zenhub.com/files/67439560/5f494500-f23d-4767-93e8-f3f57a0107db/download](https://app.zenhub.com/files/67439560/5f494500-f23d-4767-93e8-f3f57a0107db/download)

### PR DESCRIPTION
**Pull request by FloLey**
# Description

add a changelog deprecation note stating the deprecation of Jinja's is defined for nullable attributes

closes #1888 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: changelogs/unreleased/1888-depreacation-jinja-is-defined.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+description: Remove support for leaving nullable attribute unassigned. You should make sure optional variables are always assigned a value. This changes the behaviour of 'is defined' in Jinja templates. You should now use 'is not none' instead.
```
Comment (sanderr) [inline changelogs/unreleased/1888-depreacation-jinja-is-defined.yml:2]: "Remove support" is not really accurate. In iso5 we've only deprecated it, for which we already released a change entry (though it might not be a bad idea to repeat this here for context). Also "You should now use" is not completely accurate: you should only use it if you make the change to explicit `null`. Perhaps like this (I'm not sure myself, feel free to object)?
```suggestion
description: When you make the change to use explicit `null` for plain attributes (due to previous deprecation of leaving nullable attributes unassigned), the behavior of `is defined` in Jinja templates changes. You should update Jinja templates to use 'is not none' instead of `is defined` together with the change to explicit `null`.
```

**Review by sanderr — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/1888-depreacation-jinja-is-defined.yml (hunk 1 above)
Comment (FloLey) [reply changelogs/unreleased/1888-depreacation-jinja-is-defined.yml:2]: I'm not sure I understand this part: `(due to previous deprecation of leaving nullable attributes unassigned)` as it has never been mentioned before on iso5 I think?

Comment (sanderr) [reply changelogs/unreleased/1888-depreacation-jinja-is-defined.yml:2]: It appears you're right, I assumed we'd added a message when we introduced the deprecation warning. In that case your message was good, except I would rephrase "remove support for" as "deprecated" (because support will only be dropped in iso6). And optionally rephrase the "you should now use" but that's of lesser importance.

**Review by sanderr — APPROVED** [review]
